src/app/server/Server.cpp

```diff
@@ -62,11 +63,68 @@ bool isRendezvousBypassed()
     return rendezvousMode == RendezvousInformationFlags::kNone;
 }
 
+class DeviceDiscriminatorCache
+{
+public:
+    CHIP_ERROR UpdateDiscriminator(uint16_t discriminator)
+    {
+        if (!mOriginalDiscriminatorCached)
+        {
+            // Cache the original discriminator
+            ReturnErrorOnFailure(DeviceLayer::ConfigurationMgr().GetSetupDiscriminator(mOriginalDiscriminator));
+            mOriginalDiscriminatorCached = true;
+        }
+
+        return DeviceLayer::ConfigurationMgr().StoreSetupDiscriminator(discriminator);
```

Comment:
Is the idea with storing this that it survives a reboot? But we don't store that we need to restore it..

Or are we really using "configuration" mgr for temporary state?

Reply:
Yeah, the current BLE manager implementation uses configuration manager to read the discriminator. We need a better way to communicate the temporary discriminator to the BLE, so that it does not need to be persisted. That'll remove the need for this mechanism.

Reply:
Please add a TODO then.

Reply:
Done